src/hotspot/share/opto/mulnode.cpp

```diff
@@ -610,6 +610,14 @@ Node *AndINode::Ideal(PhaseGVN *phase, bool can_reshape) {
     return progress;
   }
 
+  // Convert "(~a) & (~b)" into "~(a | b)"
+  if (in(1)->Opcode() == Op_XorI
+      && phase->type(in(1)->in(2)) == TypeInt::MINUS_1
+      && in(2)->Opcode() == Op_XorI
+      && in(1)->in(2) == in(2)->in(2)) {
+    return new XorINode(phase->transform(new OrINode(in(1)->in(1), in(2)->in(1))), in(1)->in(2));
```

Comment:
The nesting of this line is difficult to read. I suggest you take multiple lines and name intermediate results with something helpful.